### ISSUE TITLE
Issue with text echo in Terminal (bash) after running sbt in 1.0.0

### ISSUE THREAD

**Issue by justinrmiller** [sbt/sbt#3453]
## steps
```
macbook:scala-playground justin$ sbt --help
[info] Loading settings from assembly.sbt,plugins.sbt ...
[info] Loading project definition from /Users/justin/Development/Projects/scala-playground/project
[info] Loading settings from build.sbt ...
[info] Set current project to scala-playground (in build file:/Users/justin/Development/Projects/scala-playground/)
[warn] The `-` command is deprecated in favor of `onFailure` and will be removed in 0.14.0
macbook:scala-playground justin$ -bash: bob: command not found
macbook:scala-playground justin$ 
```
## problem

After running the sbt command line tool, console keyboard echo output appears to stop working.

## expectation

After running sbt, I expect keyboard echo to still work without having to do "reset".

## notes

sbt version: 1.0.0

**Comment by liff** [comment]
This also happens if running non-interactively, like `sbt compile`.

**Comment by dwijnand** [comment]
As a stop gap you might want to upgrade to the latest sbt-extras, which just worked around this issue: https://github.com/paulp/sbt-extras/pull/200.

**Comment by ekrich** [comment]
I thought this was fixed (except for non-standard terminals) when I tested on one the 1.0.0-M5/M6 versions?
Reference: https://github.com/sbt/sbt/issues/2963

In general are the `stty` settings saved and then restored on sbt exit? Is the `ScriptMain` entry point different than a regular entry point in regards to shell handling?

Workaround by typing `stty echo` or by using newest version of https://github.com/paulp/sbt-extras which worked when tested.
